lib/rich-text.coffee

```diff
@@ -0,0 +1,102 @@
+Characters = require 'characters'
+Selections = require '../../y-selections'
+
+# A class holding the information about rich text
+class RichText
+  # @param content [String] an initial string
+  # @param editor [Editor] an editor instance
+  # @param author [String] the name of the local author
+  constructor: (content, editor, author) ->
+    # TODO: generate a UID
+    @selections = new Selections
+    @characters = new Characters content, @selections
+
+    @author = author
+
+    @editor = editor
+    # shouldn't go there because of modularity
+    @editor.cursors = @editor.getModule("multi-cursor")
+
+  _getModel: (Y, Operation) ->
+    if not @_model?
+      sels = @selections._getModel(Y, Operation)
+      chars = @characters._getModel(Y, Operation)
+      cursors = new Operation.ListManager(@).execute()
+
+      @_model = new Operation.MapManager(@).execute()
+      @_model.val "selections", sels
+      @_model.val "characters", chars
+      @_model.val "cursors", cursors
+
+      @setCursor()
+
+    return @_model
+
+  _setModel: (model) ->
+    @_model = model
+    # check that our cursor isn't in the cursor list
+    if _.all (@get "cursors").val(), (cursor -> cursor.author != @author)
+      @setCursor()
+
+  set: (key, val) ->
+    @_model.val key, val
+  get: (key) ->
+    @_model.val key
+  delete: (key) ->
+    @_model.delete key
+
+  setCursor: () ->
+    if @editor.getSelection()
+      position = @characters.val @editor.getSelection().start
+    else
+      position = @characters.val 0
+
+    selfCursor =
+      author: @author
+      position: position
+      color: "grey" # FIXME
+    (@get "cursors").insert 0, selfCursor
+
+  toQuill: (events) ->
```

Comment:
toQuill is a bad method name because:
1. it does not explain what this method does
2. y-richtext should not be editorspecific (so we have to change his in the future)

This should rather be a "private" (therefore scoped) function (not method), that is not accessible to third party users.